gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
I think both the EnergyOffsetArray class and the container class that contains three arrays should have read / write.
I.e. keep the read / write here as-is!

The container class then calls the to_fits for each array and puts the three HDUs in one FITS file with extension names like "counts", "livetime" and "rate" (or whatever the 3D cube class  does).

Does this make sense?

(You could already start the pull request for the container class now, and for now just do the rate computation there and I/O, and then we add other stuff as needed in the coming days ...)